src/components/Comments/Comments.jsx

```diff
@@ -1,35 +1,76 @@
-import { useEffect, useState } from "react";
-import { getAllComments } from "../../apis/apis";
+import { useEffect, useState, useContext } from "react";
+import { deleteAComment, getAllComments } from "../../apis/apis";
 import { useParams } from "react-router-dom";
 import { Fade } from "react-awesome-reveal";
 import HourglassTopIcon from "@mui/icons-material/HourglassTop";
+import CommentAdder from "./CommentAdder";
+import DeleteIcon from "@mui/icons-material/Delete";
+import { UserContext } from "../Context/usersContext";
 
 function Comments() {
   const { id } = useParams();
   const [comments, setComments] = useState([]);
   const [loadingComments, setLoadingComments] = useState(true);
+  const { user } = useContext(UserContext);
 
   useEffect(() => {
     getAllComments(id).then((res) => {
-      setComments(res.data);
       setLoadingComments(false);
+      setComments(res.data.comments);
     });
   }, []);
 
+  const handleDelete = (comment_id) => {
+    deleteAComment(comment_id).then((res) => {
+      const newComments = comments.filter(
+        (com) => com.comment_id !== comment_id
+      );
+      setComments(newComments);
+
+      if (res.status === 204) {
+        alert("Comment Deleted");
+      } else {
+        alert("Comment failed to delete");
+      }
+    });
+  };
```

Comment:
Nice 👍

Reply:
The only thing to add here is that deleting should only be able to be done by the person who posted the comment. Please add this conditional logic :)